pyqtgraph/graphicsItems/PColorMeshItem.py

```diff
@@ -66,27 +104,35 @@ def __init__(self, *args, **kwargs):
         GraphicsObject.__init__(self)
 
         self.qpicture = None  ## rendered picture for display
+        self.x = None
+        self.y = None
+        self.z = None
         
-        self.axisOrder = getConfigOption('imageAxisOrder')
-
-        if 'edgecolors' in kwargs.keys():
-            self.edgecolors = kwargs['edgecolors']
-        else:
-            self.edgecolors = None
-
-        if 'antialiasing' in kwargs.keys():
-            self.antialiasing = kwargs['antialiasing']
-        else:
-            self.antialiasing = False
+        self.edgecolors = kwargs.get('edgecolors', None)
+        self.antialiasing = kwargs.get('antialiasing', False)
         
-        if 'cmap' in kwargs.keys():
-            if kwargs['cmap'] in Gradients.keys():
-                self.cmap = kwargs['cmap']
-            else:
+        if 'colorMap' in kwargs:
+            cmap = kwargs.get('colorMap')
+            if not isinstance(cmap, colormap.ColorMap):
+                raise ValueError('colorMap argument must be a ColorMap instance')
+            self.cmap = cmap
+        elif 'cmap' in kwargs:
```

Comment:
Thanks for including this @pijyoi the only question that I have here is do we want to emit a deprecation warning?

Reply:
Okay, I have added a deprecation warning with a runway of 1 year.
But generally, I think it would be sufficient to provide a new API that is more featureful to gently nudge users over. e.g. pg.ColorMap has way more colors. Although in this case, PColorMeshItem is fairly new, so shouldn't have many users to begin with.